ambry-store/src/main/java/com.github.ambry.store/CompactionManager.java

```diff
@@ -301,6 +314,16 @@ boolean scheduleNextForCompaction(BlobStore store) {
       }
       return true;
     }
+
+    /**
+     * Disable the compaction on given blobstore
+     * @param store the {@link BlobStore} on which the compaction is disabled.
+     */
+    void disableCompactionForBlobStore(BlobStore store) {
+      if (store.isStarted()) {
+        storesToSkip.add(store);
```

Comment:
storesToSkip is a hashset, which is not thread safe. If you are planning on calling this from AmbryRequests, you might have to switch to a thread safe collection.

Reply:
+1

Also I think we should add whether or not the store is started.